File: from-2201-to-2300/2257-count-unguarded-cells-in-the-grid/solution.py
class Solution:
    def countUnguarded(self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]) -> int:
        grid = [[0] * n for _ in range(m)]
        # 0 - free, 1 - guard, 2 - wall, 3 - guardable

        for r, c in guards:
            grid[r][c] = 1
        for r, c in walls:
            grid[r][c] = 2

        for r, c in guards:
            for i in range(r - 1, -1, -1):
                if grid[i][c] == 2 or grid[i][c] == 1:
                    break
                grid[i][c] = 3
            for i in range(r + 1, m, 1):
                if grid[i][c] == 2 or grid[i][c] == 1:
                    break
                grid[i][c] = 3
            for j in range(c - 1, -1, -1):
                if grid[r][j] == 2 or grid[r][j] == 1:
                    break
                grid[r][j] = 3
            for j in range(c + 1, n, 1):
                if grid[r][j] == 2 or grid[r][j] == 1:
                    break
                grid[r][j] = 3

        result = 0
        for r in range(m):
            for c in range(n):
                if grid[r][c] == 0:
                    result += 1

        return result

File: from-2201-to-2300/2257-count-unguarded-cells-in-the-grid/solution.go
func countUnguarded(m int, n int, guards [][]int, walls [][]int) int {
    grid := make([][]int, m)
    for i := 0; i < m; i ++ {
        grid[i] = make([]int, n)
    }
    // 0 - free, 1 - guard, 2 - wall, 3 - guardable

    for _, g := range guards {
        grid[g[0]][g[1]] = 1
    }
    for _, w := range walls {
        grid[w[0]][w[1]] = 2
    }

    for _, g := range guards {
        r, c := g[0], g[1]
        for i := r - 1; i >= 0; i -- {
            if grid[i][c] == 1 || grid[i][c] == 2 {
                break
            }
            grid[i][c] = 3
        }
        for i := r + 1; i < m; i ++ {
            if grid[i][c] == 1 || grid[i][c] == 2 {
                break
            }
            grid[i][c] = 3
        }
        for j := c - 1; j >= 0; j -- {
            if grid[r][j] == 1 || grid[r][j] == 2 {
                break
            }
            grid[r][j] = 3
        }
        for j := c + 1; j < n; j ++ {
            if grid[r][j] == 1 || grid[r][j] == 2 {
                break
            }
            grid[r][j] = 3
        }
    }

    result := 0
    for r := 0; r < m; r ++ {
        for c := 0; c < n; c ++ {
            if grid[r][c] == 0 {
                result += 1
            }
        }
    }
    return result
}